<!DOCTYPE html>
<html lang="en">
<head>
<title>Skesa</title>
<link rel="stylesheet" type="text/css" href="tutorial_stylesheet.css">
</head>
<body>
<h1><center>GalaxyTrakr Basics: FASTQ Import</center></h1> 
<div class="sidenav">
  <p>GalaxyTrakr Basics</p>
  <a href="https://microseqtrain.github.io/fastq_import">FASTQ Import</a>
  <a href="https://microseqtrain.github.io/Paired_fastq_collection">Paired FASTQ Collection</a>
  <a href="https://microseqtrain.github.io/Workflow_import">Workflow Import</a>
  <p>Tool Tutorials</p>
  <a href="https://microseqtrain.github.io/skesa">Skesa</a>
  <a href="https://microseqtrain.github.io/FastQC">FastQC</a>
  <a href="https://microseqtrain.github.io/MicroRunQC">MicroRunQC</a>
  <a href="https://microseqtrain.github.io/MLST">MLST</a>
  <a href="https://microseqtrain.github.io/AMRFinderPlus">AMRFinderPlus_db</a>
  <a href="https://microseqtrain.github.io/SeqSero2">SeqSero2</a>
  <a href="https://microseqtrain.github.io/MicroRunQC">MicroRunQC</a>
  <a href="https://microseqtrain.github.io/index">About Us</a>
</div>
<p><center>Skesa is a tool for creating an assembly from fastq or fasta files that have breaks in the repeat sections of the genome.</center></p>
<button class="collapsible">Video Tutorial</button>
<div class="content">
	<p><i>Coming soon!</i></p>
</div>
<button class="collapsible">Text Walkthrough: From Network/User's Computer</button>
<div class="content">
	<p><i>If you are importing your data from your local network or your own computer, please perfom the following:.</i></p>
	<p>After sigining in to GalaxyTrakr, make sure you are in the History where you would like your files uploaded then click on the button that reads <q>Upload Data.</q> Click <q>Choose local files</q> and navigate to the folder that contains your files of interest, select those files, and click 
		<q>Open.</q> The file names will now appear in the pane in GalaxyTrakr, along with their size and options for their import. We would suggest leaving Type as the <q>Auto-detect</q> default. You do not need to select an 
		organism type under the <q>Genome</q> column unless you are certain that the sequence you are uploading is from a pure sample and you know its ID. If you have selected a file for upload by mistake, remove it from
		the upload by clicking on the trashcan icon to the right of it.</p>
	<p>Once you have your settings the way you want them, click <q>Start</q> to begin the upload. The Status bars for the files will begin filling. The pane does not have to remain open for the upload to continue. Click the
		<q>Close</q> button to return to the GalaxyTrakr main page. The progress of the upload can also be seen in the same <q>Upload</q> data button that was used to initiate the upload. Click that button again if you need
		to open that pane again. The upload can be paused by doing so and clicking the <q>Pause</q> button.</p>
	<p>The files will appear in the History once the upload is complete.</p>

</div>
<button class="collapsible">Text Walkthrough: From NCBI</button>
<div class="content">
	<p><i>If you are importing data from NCBI without using your computer as an intermediary, please follow these steps.</i></p>
	<p>After signing in to GalaxyTrakr, make sure you are in teh History where you would like your files uploaded, then search for <q>NCBI</q> in the Tool search bar at the left hand side of the main GalaxyTrakr page. You are
		looking for the tool called <q>Faster Download and Extract Reads in FASTQ.</q> Click on it. You have three options for input in this tool:</p>
		<il>
		<li><b>SRR accession</b>, where you can input a single SRR that will have its FASTQ files added to the History.</li>
		<li><b>List of SRA accession, one per line</b>, where you can input list of SRRs that will have their FASTQ files added to the History. Please make sure these are in line-by-line format rather than a comma separated 
		list.</li>
		<li><b>SRA archive in current history</b>, where an SRR accession list in a file uploaded to your History will be converted into a list of FASTQ files. Please see the documentation at the bottom of the tool's page
		in GalaxyTrakr for further details of this method.</li>
		</il>
	<p>Advanced options in the tool will not apply in most use cases. Please contact us at <a href="mailto:genometrakr@fda.hhs.gov">microseqtrain@fda.hhs.gov</a> if you would like to discuss use of those options.
		<q>Email notification</q> at the bottom of the page if you would like GalaxyTrakr to email you when the data has been completely extracted from NCBI. When the options are as you wish, click <q>Execute</q> to begin the
		extraction from NCBI.</p>

</div>
<button class="collapsible">References</button>
<div class="content">
	<p>Documentation for Skesa can be found <a href="https://github.com/ncbi/ngs-tools/tree/master/tools/skesa/">here</a> on the developer's GitHub page.</p>
</div>

</body>
</html>