<!DOCTYPE html>
<html lang="en">
<head>
<title>MicroRunQC</title>
<link rel="stylesheet" type="text/css" href="tutorial_stylesheet.css">
</head>
<body>
<div class="sidenav">
  <p>GalaxyTrakr Basics</p>
  <a href="https://microseqtrain.github.io/fastq_import">FASTQ Import</a>
  <a href="https://microseqtrain.github.io/Paired_fastq_collection">Paired FASTQ Collection</a>
  <a href="https://microseqtrain.github.io/Workflow_import">Workflow Import</a>
  <p>Tool Tutorials</p>
  <a href="https://microseqtrain.github.io/skesa">Skesa</a>
  <a href="https://microseqtrain.github.io/FastQC">FastQC</a>
  <a href="https://microseqtrain.github.io/MicroRunQC">MicroRunQC</a>
  <a href="https://microseqtrain.github.io/MLST">MLST</a>
  <a href="https://microseqtrain.github.io/AMRFinderPlus">AMRFinderPlus_db</a>
  <a href="https://microseqtrain.github.io/SeqSero2">SeqSero2</a>
  <a href="https://microseqtrain.github.io/index">About Us</a>
</div>

<h1><center>MicroRunQC</center></h1> 
<p><center>MicroRunQC is a workflow in GalaxyTrakr that takes raw fastq files, runs the tools Trimmomatic and microrunqc, and concatenates and filters the resulting output into collections of draft <i>de novo</i>
			assemblies, quality assessments, and sequencing types. The latter are contained in tabular data files for download. The protocol linked below in the <q>References</q> section covers use of the Workflow
			in detail.</center></p>
<button class="collapsible">Video Tutorial</button>
<div class="content">
	<p><i>Coming soon!</i></p>
</div>

<button class="collapsible">References</button>
<div class="content">
	<p>Documentation for Skesa can be found <a href="https://www.protocols.io/view/quality-control-assessment-for-microbial-genomes-g-4r3l27oz3g1y/v1">in GenomeTrakr's protocols.io resources</a>.</p>
</div>

</body>
</html>