<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="UTF-8"/>
	<meta name="viewport"
	content="width=device-width, initial-scale=1.0"/>
<title>MicroSeqTrain | MLST Tutorial </title>
<link rel="stylesheet" href="tutorial_stylesheet.css"/>
</head>

<body>

	<div class="banner">
		<h1>MicroSeqTrain</h1>
	</div>
	
<!-- This section is the side menu bar. Do not edit it unless adding or changing a link to another page on the website. -->
<div class="container">
	<nav class="sidebar">
		<p>GalaxyTrakr Basics</p>
		<ul>
			<li><a href="https://microseqtrain.github.io/histories">Histories</a></li>
			<li><a href="https://microseqtrain.github.io/fastq_import">FASTQ Import</a></li>
			<li><a href="https://microseqtrain.github.io/paired_fastq_collection">Paired FASTQ Collection</a></li>
			<li><a href="https://microseqtrain.github.io/workflow_import">Workflow Import</a></li>
			<li><a href="https://microseqtrain.github.io/basic_troubleshooting">Basic Troubleshooting</a></li>
		</ul>
		<br>
		<p>Short Read Processing</p>
		<ul>
			<li><a href="https://microseqtrain.github.io/skesa">Skesa</a></li>
			<li><a href="https://microseqtrain.github.io/FastQC">FastQC</a></li>
			<li><a href="https://microseqtrain.github.io/MicroRunQC">MicroRunQC</a></li>
			
		</ul>
		<br>
		<p>Long Read Processing</p>
		<ul>
			<li><a href="https://microseqtrain.github.io/filtlong">filtlong</a></li>
			<li><a href="https://microseqtrain.github.io/Flye">Flye</a></li>
			<li><a href="https://microseqtrain.github.io/NanoPlot">NanoPlot</a></li>
		</ul>
		<br>
		<p>Analysis Tools</p>
		<ul>
			<li><a href="https://microseqtrain.github.io/MLST">MLST</a></li>
			<li><a href="https://microseqtrain.github.io/AMRFinderPlus_db">AMRFinderPlus_db</a></li>
			<li><a href="https://microseqtrain.github.io/Kraken2">Kraken2</a></li>
			<li><a href="https://microseqtrain.github.io/SeqSero2">SeqSero2</a></li>
		</ul>
		<br>
		<a href="https://microseqtrain.github.io/about">About Us</a>
		
		<br>
		
		<div class="search-container">
			<input type="text" placeholder="Search..." class="search-bar" />
		</div>
	</nav>
<!-- End sidebar menu code. -->

<main class="content">

	<h2><center>MLST</center></h2>

	<section id="description">
		<p>MLST is a tool that will run input genomes against PubMLST schemes and output the scheme most closely matching the input genome along with the alleles used to select the scheme.</p>
		<p><b>Inputs:</b> fasta or Genbank sequencing files. If your raw data is in fastq format, it is advised that the data be run through an assembler, such as <a href = "https://microseqtrain.github.io/Skesa">Skesa</a> 
			to both improve quality and put the data in assembled, fasta format.</p>
		<p><b>Output: </b> A single tabular data file containing the filenames of your input files, the scheme found to match each input file, and the alleles used to match the scheme to the input.</p>
	</section>
	
<br>
	
	<section class="collapsible-section">
		<input type="checkbox" id="collapse1" />
		<label for="collapse1" class="collapsible-label">Operation</label>
		<div class="collapsible-content" id="operation">
			<p>For the purposes of this tutorial, it is assumed that the user has previously created a history, uploaded fastq or fasta files, and run an assembler such as <a href = "https://microseqtrain.github.io/Skesa">Skesa</a>. 
				If you need help with the history and file import would like to learn more about those steps, please visit <a href="https://microseqtrain.github.io/histories">this</a> tutorial for help with histories 
				and <a href="https://microseqtrain.github.io/fastq_inport">this</a> tutorial for fastq inport.</p>
			<p>Once you have your assembled data in your history, search for <q>MLST</q> in the <b>search tools</b> bar and select the <b>MLST</b> tool.</p> 
			<p>Input either the multiple datasets containing your FASTA files or a collection of such files. Note that when the multiple datasets option is used, each sequence to be filtered must be selected as if
				selecting multiple files in Windows (hold down the <q>ctrl</q> key and click on items in the list individually or hold down the <q>shift</q> key and highlight the last item in
				a group to be selected).</p> 
			<p>If you do not desire to specify advanced parameters, click <b>Execute</b> to run the tool. If you have reason to believe you need to alter the default parameters for your particular use case, feel free to 
				reach out to us at <a href="mailto:microseqtrain@fda.hhs.gov">microseqtrain@fda.hhs.gov</a> if you have any questions about doing so.</p>
		</div>
	</section>
	
	<br>
	
	<section class="collapsible-section">
		<input type="checkbox" id="collapse2" />
		<label for="collapse2" class="collapsible-label">References</label>
		<div class="collapsible-content" id="references">
			<p>Documentation for MLST can be found at these locations:</p>
			<ul>
				<li><a href="https://github.com/tseemann/mlst"> T, S. (2016). MLST: Scan contig files against PubMLST typing schemes.</a></li>
				<li><a href="https://bio.tools/mlst">bio.tools</a></li>
				<li><a href="https://openebench.bsc.es/tool/mlst">OpenEBench</a></li>
		</div>
	</section>
	
	<br>
	
	<section class="collapsible-section">
		<input type="checkbox" id="collapse3" />
		<label for="collapse3" class="collapsible-label">Video Walkthrough</label>
		<div class="collapsible-content" id="video">
			<p>This is where the video tutorial will live.</p>
		</div>
	</section>
	
	<br>
	
	<section class="collapsible-section">
		<input type="checkbox" id="collapse4" />
		<label for="collapse4" class="collapsible-label">Walkthrough Example Downloads</label>
		<div class="collapsible-content" id="example-downloads">
			<p>The video tutorial started with a collection of 10 GenomeTrakr Validation Strain Set sequence pairs that had been run through Skesa. To follow along, begin by
			importing these same files from NCBI by downloading this <a href="galaxytrakr_tutorial_10_miseq.txt" download="download">txt file</a> containing the list of SRR 
			accessions for the 10 sequences, uploading the file into GalaxyTrakr and using it to import the sequencing data from NCBI (see <a href="https://microseqtrain.github.io/fastq_import">this tutorial</a>
			for assistance), and running <a href="https://microseqtrain.github.io/skesa">Skesa</a> on the resulting collection of sequences.</p>
			<p><a href="MLST_output_example.tabular" download="MLST_output_example.tabular">This file</a> contains the output from running MLST on the Skesa results.
		</div>
	</section>
</main>
</div>
 </body>

 </html>
