<!DOCTYPE html>
<html lang="en">

<head>
	<meta charset="UTF-8"/>
	<meta name="viewport"
	content="width=device-width, initial-scale=1.0"/>
<title>MicroSeqTrain | NanoPlot Tutorial</title>
<link rel="stylesheet" href="tutorial_stylesheet.css"/>
</head>

<body>

	<div class="banner">
		<h1>MicroSeqTrain</h1>
	</div>

<div class="container">
	<nav class="sidebar">
		<p>GalaxyTrakr Basics</p>
		<ul>
			<li><a href="https://microseqtrain.github.io/histories">Histories</a></li>
			<li><a href="https://microseqtrain.github.io/fastq_import">FASTQ Import</a></li>
			<li><a href="https://microseqtrain.github.io/paired_fastq_collection">Paired FASTQ Collection</a></li>
			<li><a href="https://microseqtrain.github.io/workflow_import">Workflow Import</a></li>
			<li><a href="https://microseqtrain.github.io/basic_troubleshooting">Basic Troubleshooting</a></li>
		</ul>
		<br>
		<p>Short Read Processing</p>
		<ul>
			<li><a href="https://microseqtrain.github.io/skesa">Skesa</a></li>
			<li><a href="https://microseqtrain.github.io/FastQC">FastQC</a></li>
			<li><a href="https://microseqtrain.github.io/MicroRunQC">MicroRunQC</a></li>
			
		</ul>
		<br>
		<p>Long Read Processing</p>
		<ul>
			<li><a href="https://microseqtrain.github.io/filtlong">filtlong</a></li>
			<li><a href="https://microseqtrain.github.io/Flye">Flye</a></li>
			<li><a href="https://microseqtrain.github.io/NanoPlot">NanoPlot</a></li>
		</ul>
		<br>
		<p>Analysis Tools</p>
		<ul>
			<li><a href="https://microseqtrain.github.io/MLST">MLST</a></li>
			<li><a href="https://microseqtrain.github.io/AMRFinderPlus">AMRFinderPlus_db</a></li>
			<li><a href="https://microseqtrain.github.io/Kraken2">Kraken2</a></li>
			<li><a href="https://microseqtrain.github.io/SeqSero2">SeqSero2</a></li>
		</ul>
		<br>
		<a href="https://microseqtrain.github.io/about">About Us</a>
		
		<br>
		
		<div class="search-container">
			<input type="text" placeholder="Search..." class="search-bar" />
		</div>
	</nav>
	
<main class="content">

	<h2><center>NanoPlot</center></h2>

	<section id="description">
		<p>NanoPlot is a GalaxyTrakr tool for visualizing sequencing data and alignments resulting from Oxford Nanopore Technologies (ONT) long read sequences.</p>
		<p><b>Inputs:</b> fastq, fasta, fastq_rich, fastq_minimal, summary, bam, or cram file(s) from ONT or subsequent sequence processing. <b>NOTE:</b>If you have reason to believe your 
			sequencing data is of questionable quality, it would be wise to run sequences through a filtering tool, like filtlong, first. See tutorial for filtlong <a href="https://microseqtrain.github.io/filtlong">here</a>.</p>
		<p><b>Outputs:</b></p>
		<ul>
			<li>Text <i>NanoStats</i> (pre-filtering) file containing summary statistics about the quality of the file(s) analyzed.</li>
			<li>Text <i>NanoStats post filtering</i> file containing summary statistics about the quality of the file(s) analyzed after they have been filtered.</li>
			<li>png or svg file(s) containing a <i>histogram visualizing read length data</i>.</li>
			<li>png or svg file(s) containing a <i>histogram visualizing read length data after it has undergone a log transformation</i>.</li></li>
			<li>An <i>html</i> file that contains all of the summary data and plots generated by the tool.</li>
		</ul>
	</section>
	
<br>
	
	<section class="collapsible-section">
		<input type="checkbox" id="collapse1" />
		<label for="collapse1" class="collapsible-label">Operation</label>
		<div class="collapsible-content" id="operation">
			<p>This tutorial assumes that you have already imported fastq or fasta file(s) from a long read sequencer into a GalaxyTrakr history. If you need assistance with that task, please see 
				<a href="https://microseqtrain.github.io/fastq_import">this</a> tutorial.</p>
			<p>Search for and click on <q>NanoPlot</q> in the tool search bar of GalaxyTrakr.</p>
			<p>You have the option of running NanoPlot on an individual sequencing file (select single dog-eared sheet symbol), multiple sequencing files that have not previously been a collection 
				(select the icon with multiple sheets), and a collection of sequencing files (for instructions on how to make such a collection, visit <a href="https://microseqtrain.github.io/paired_fastq_collection">this</a> 
				tutorial. Note that when the multiple datasets option is used, each sequence to be analyzed must be selected as if selecting multiple files in Windows (hold down the <q>ctrl</q> key and click on items 
				in the list individually or hold down the <q>shift</q> key and highlight the last item in a group to be selected).</p>
			<p><b>Note:</b>Unlike some other tools, NanoPlot will give you the option of whether to run the tool on multiple files with their results appearing in separate data files and
				plots (<b>batch</b>) or with the results in a single file for each output type (<b>combined</b>) in the <b>Select multifile mode</b> dropdown menu.</p>
			<p>It is not necessary to change the modifiers under <b>Options for filtering or transforming input prior to plotting</b>. However, you can use these options to dictate how
				NanoPlot filters data prior to assessing quality. Email us at <a href="mailto:microseqtrain@fda.hhs.gov"></a> if you would like to discuss how these options may be beneficial in 
				your particular use case. There is also a toggle under this section that should be switched if the tool will be analyzing <b>bam</b> files.</p>
			<p>There is also an optional section for changing the appearance of the graphs NanoPlot produces and other output attributes, <b>Options for customizing the plots created</b>.
				Attributes that can be changed include the color of the plots, the bivariate format(s) of the plots, and whether or not N50 is marked on the read length histogram.</li>
				<b>Note:</b>If you would like to choose svg format instead of the default png format for the histograms NanoPlot produces, the <b>Specify the output format of the plots.</b>
				dropdown in this section is where you would do it.
			<p>Choose whether you would like to be emailed when NanoPlot is finished running using the toggle, and click <q>Execute</q> to run the tool.</p>
		</div>
	</section>
	
	<br>
	
	<section class="collapsible-section">
		<input type="checkbox" id="collapse2" />
		<label for="collapse2" class="collapsible-label">References</label>
		<div class="collapsible-content" id="references">
			<p>Documentation for NanoPlot can be found at these sources: 
			<ul>
				<li>De Coster, W., D’Hert, S., Schultz, D. T., Cruts, M., & Van Broeckhoven, C. (2018). NanoPack: visualizing and processing long-read sequencing data. 
					<i>Bioinformatics</i>, 34(15), 2666–2669. Retrieved from <a href="http://dx.doi.org/10.1093/bioinformatics/bty149">http://dx.doi.org/10.1093/bioinformatics/bty149</a></li>
				<li>On the GitHub page of one of the developers, W. De Coster, <a href="https://github.com/wdecoster/NanoPlot">here</a>.</li>
			</ul>  
		</div>
	</section>
	
	<br>
	
	<section class="collapsible-section">
		<input type="checkbox" id="collapse3" />
		<label for="collapse3" class="collapsible-label">Video Walkthrough</label>
		<div class="collapsible-content" id="video">
			<p>This is where the video tutorial will live.
		</div>
	</section>
	
	<br>
	
	<section class="collapsible-section">
		<input type="checkbox" id="collapse4" />
		<label for="collapse4" class="collapsible-label">Walkthrough Example Downloads</label>
		<div class="collapsible-content" id="example-downloads">
			<p><a href="ONT_seq_examples.zip" download="download">This</a> zip file contains the ONT sequences used in the video tutorial. In order to import them into 
				GalaxyTrakr as quickly as possible, it would be best to extract the files from the folder, compress them into zip files individually, and then upload them into
				GalaxyTrakr as instructed in the <a href="https://microseqtrain.github.io/fastq_import">FASTQ Import</a> tutorial. After import, begin using this tutorial's
				instructions.</p>
		</div>
	</section>
</main>
</div>
 </body>
 </html>