<!DOCTYPE html>
<html lang="en">
<head>
<title>SeqSero2</title>
<link rel="stylesheet" type="text/css" href="tutorial_stylesheet.css">
</head>
<body>
<h1><center>SeqSero2</center></h1>
<div class="sidenav">
  <p>GalaxyTrakr Basics</p>
  <a href="https://microseqtrain.github.io/fastq_import">FASTQ Import</a>
  <a href="https://microseqtrain.github.io/paired_fastq_collection">Paired FASTQ Collection</a>
  <a href="https://microseqtrain.github.io/workflow_import">Workflow Import</a>
  <br>
  <p>Tool Tutorials</p>
  <a href="https://microseqtrain.github.io/skesa">Skesa</a>
  <a href="https://microseqtrain.github.io/FastQC">FastQC</a>
  <a href="https://microseqtrain.github.io/MicroRunQC">MicroRunQC</a>
  <a href="https://microseqtrain.github.io/MLST">MLST</a>
  <a href="https://microseqtrain.github.io/AMRFinderPlus">AMRFinderPlus_db</a>
  <a href="https://microseqtrain.github.io/SeqSero2">SeqSero2</a>
  <br>
  <a href="https://microseqtrain.github.io/index">About Us</a>
</div> 
<p><center>SeqSero2 is a tool used to predict the serotype of <i>Salmonella</i> isolates.</center></p>
<button class="collapsible">Video Tutorial</button>
<div class="content"> 
	<p><i>Coming soon!</i></p>
</div>
<button class="collapsible">Text Walkthrough</button>
<div class="content">
	<p>You may begin with a paired collection of forward and reverse FASTQ files, two datasets containing forward and reverse FASTQ files, or FASTA files. For assistance with uploading data, please see the GalaxyTrakr Basics 
	tutorials.</p>
	<p>Once you have your data in your history, search for <q>SeqSero2</q> in the <q>search tools</q> bar and select the <q>SeqSero2</q> tool.</p>
	<p>Select your chosen dataset type from the dropdown of the tool. If you have a dataset of the type selected, it will autopopulate in the next dropdown. If this is not the input you would like to use, use the dropdown to 
	select the correct dataset.</p>
	<p>For <q>Analysis mode,</q> allele mode was what was used to evaluate SeqSero2 by the GenomeTrakr team. Please contact us if you would like to discuss whether k-mer mode would be better for you.</p>
	<p>Leave the dropdown for Algorithms for BWA mapping at the default, which is mem. This is the only algorithm for which SeqSero2 is currently optimized.</p>
	<p>Once you are happy with the inputs, click <q>Execute</q> to run the tool.</p>
	<p>SeqSero2 will create an output containing a table for each sequence uploaded detailing the O and H antigens coded for, the predicted identification of the Salmonella species, the predicted antigenic profile, the predicted
	serotype, potential inter-serotype contamination, if any, and any notes about the sequence, including whether the sequence uploaded does not align with Salmonella genomes.</p>


</div>
<button class="collapsible">References</button>
<div class="content">
	<p>Documentation for SeqSero2 can be found <a href="https://github.com/CFSAN-Biostatistics/SeqSero2-galaxy">here</a> on the developer's GitHub page.</p>
</div>

</body>
</html> 